reflex/app.py

```diff
@@ -714,8 +719,10 @@ async def modify_state(self, token: str) -> AsyncIterator[State]:
         """
         if self.event_namespace is None:
             raise RuntimeError("App has not been initialized yet.")
+
+        self.validate_state_manager()
         # Get exclusive access to the state.
-        async with self.state_manager.modify_state(token) as state:
+        async with self.state_manager.modify_state(token) as state:  # type: ignore
```

Comment:
is this `type: ignore` because `state_manager` could possibly be None? 

It would be nice if we could find a workaround for that... like maybe change `state_manager` to `_state_manager` (and make it a `pydantic.PrivateAttr`), and then have `state_manager` be a `property` that always returns a `StateManager` instance or raises an exception?

With this approach, we could get rid of the explicit `validate_state_manager` calls and all the other bits of test code that have to `assert app.state_manager is not None` to avoid typing errors.

Reply:
This is a better approach, got it.